src/springfield/entity.py

```diff
@@ -105,9 +107,11 @@ def jsonify(self):
         Return a dictionary suitable for JSON encoding.
         """
         data = {}
-        for key, val in self.__values__.iteritems():
```

Comment:
The above usage examples for `flatten` and `jsonify` put to question whether defaults are a good thing in general. Currently defaults are supposed to work only for an active `Entity` object. For example:

```
user = UserEntity(name='foo')
assert user.enabled is True
assert user.created
```

However, that's not a good thing for many cases. What if you wanted a sparse entity? This is usually the case when doing partial selects. 

So for `user = UserEntity.adapt(sql('SELECT id, user FROM user WHERE id = :id'))` I would expect that `user` has exactly 2 fields, `id` and `user` and `created` and `enabled` would be Empty-ish. However because of defaults both would be populated with defaults. The only reason this doesn't happen today is because defaults are broken. This might actually be a good thing that they are broken. This same problem occurs if you do something like `UserEntity.adapt(data_from_http_request_body)` as the entity will have more data than was actually sent in the request.

So thinking it through I don't think `flatten(include_defaults=True)` is good, nor is having defaults on attributes that aren't set. What I think would be better is to have `user = UserEntity(_defaults=True)` or `user = UserEntity(); user.populate_defaults()` so that defaults are only included at entity creation time and only if you want them to be. So if you're pulling info from the DB you only get exactly the fields returned by the DB. If you're getting sparse data from a JSON body the entity will only have that data. But when you are creating a brand new entity for the first time you can do `populate_defaults()` and it will have all the defaults only the first time.